### PR TITLE
feat: [GPR-483][Sale Widget] Ability to set disabled payment methods

### PR DESCRIPTION
**Pull request by jhesgodi**
# Summary
Ability to set disabled payment methods

```typescript
// disable only coin payments
saleWidget.mount({ ..., disabledPaymentTypes=[SalePaymentTypes.CRYPTO] });
// disable all card payments
saleWidget.mount({ ..., disabledPaymentTypes=[SalePaymentTypes.DEBIT] });
saleWidget.mount({ ..., disabledPaymentTypes=[SalePaymentTypes.DEBIT, SalePaymentTypes.CREDIT] });
```
image

# TODO:
Add validation to prevent disabling all payment methods. Ideally do with typescript.

### CODE REVIEW

File: packages/checkout/widgets-sample-app/src/App.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ function App() {
         <div>
           <a
             href={
-              "/sale?amount=1&environmentId=249d9b0b-ee16-4dd5-91ee-96bece3b0473&collectionName=Iguanas"
+              "/sale?amount=1&environmentId=249d9b0b-ee16-4dd5-91ee-96bece3b0473&collectionName=Iguanas&&disabledPaymentTypes=credit"
```
Comment (dreamoftrees) [inline packages/checkout/widgets-sample-app/src/App.tsx:33]: Looks like double &&
```suggestion
              "/sale?amount=1&environmentId=249d9b0b-ee16-4dd5-91ee-96bece3b0473&collectionName=Iguanas&disabledPaymentTypes=credit"
```

File: packages/checkout/sdk/src/widgets/definitions/parameters/sale.ts [hunk 2]
```diff
@@ -23,6 +24,8 @@ export type SaleWidgetParams = {
   walletProviderName?: WalletProviderName;
   /** The language to use for the sales widget */
   language?: WidgetLanguage;
+  /** The disabled payment types */
+  disabledPaymentTypes: SalePaymentTypes[];
```
Comment (dreamoftrees) [inline packages/checkout/sdk/src/widgets/definitions/parameters/sale.ts:28]: Just a note in terms of naming, for filtering wallets I'm using `includeWallets` and `excludeWallets` .. was just wondering out loud if it made sense here to align the language like `excludePaymentTypes` or `includePaymentTypes`?

**Review by dreamoftrees — APPROVED** [review]
Great work @jhesgodi just a small fix and a comment on naming.

**Review by mimi-imtbl — APPROVED** [review]

File: packages/checkout/sdk/src/widgets/definitions/parameters/sale.ts (hunk 2 above)
Comment (jhesgodi) [reply packages/checkout/sdk/src/widgets/definitions/parameters/sale.ts:28]: Yes, it does thank you!

Comment (jhesgodi) [reply packages/checkout/sdk/src/widgets/definitions/parameters/sale.ts]: Done

**Review by dreamoftrees — APPROVED** [review]
👍 Nice work @jhesgodi

**Review by mimi-imtbl — APPROVED** [review]

**Review by mimi-imtbl — APPROVED** [review]

**Review by sharonsheah — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
👍

**Pull request merged**